### PR TITLE
Make saving to session optional

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #83, #85

### CODE REVIEW

**Comment by vjik** [comment]
Maybe create storage interface? It's will simplify `Locale` class.

Instead 

```php
  public function __construct(
        private EventDispatcherInterface $eventDispatcher,
        private UrlGeneratorInterface $urlGenerator,
        private LoggerInterface $logger,
        private ResponseFactoryInterface $responseFactory,
        array $supportedLocales = [],
        private array $ignoredRequestUrlPatterns = [],
        private ?SessionInterface $session = null,
        private bool $secureCookie = false,
        private ?DateInterval $cookieDuration = null,
    ) {
```

will:

```php
  public function __construct(
        private EventDispatcherInterface $eventDispatcher,
        private UrlGeneratorInterface $urlGenerator,
        private LoggerInterface $logger,
        private ResponseFactoryInterface $responseFactory,
        array $supportedLocales = [],
        private array $ignoredRequestUrlPatterns = [],
        private ?LocaleStorageInterface $storage = null,
    ) {
```

All parameters and code of cookie and session will be moved to `LocaleStorageInterface`  implementation.

**Comment by arogachev** [comment]
@vjik Good idea.

File: src/Locale.php [hunk 1]
```diff
@@ -35,7 +37,7 @@ final class Locale implements MiddlewareInterface
     private const DEFAULT_LOCALE_NAME = '_language';
     private const LOCALE_SEPARATORS = ['-', '_'];
 
-    private bool $saveLocale = true;
+    private bool $saveLocale = false;
```
Comment (samdark) [inline src/Locale.php:40]: Since we now can turn on/off session saving and cookie saving, can we remove general `withSaveLocale()` option?

Comment (samdark) [reply src/Locale.php:40]: https://github.com/yiisoft/yii-middleware/issues/83

**Review by samdark — APPROVED** [review]

**Pull request merged**